docker image crashes if transmission is down

Hello,
i'm running transmisison-exporter in docker from the `metalmatze/transmission-exporter` image, but if prometheus trying to scrape the exporter when transmission is down (f.e. during a restart) the container crashes with this error:

```
2020/01/09 16:50:44 failed to get torrents: Post http://localhost:9091/transmission/rpc: dial tcp 127.0.1.1:9091: connect: connection refused                                                                                              
2020/01/09 16:50:44 failed to get session stats: Post http://localhost:9091/transmission/rpc: dial tcp 127.0.1.1:9091: connect: connection refused                                                                                         
2020/01/09 16:50:44 failed to get session: Post http://localhost:9091/transmission/rpc: dial tcp 127.0.1.1:9091: connect: connection refused                                                                                               
panic: runtime error: invalid memory address or nil pointer dereference                                                                                                                                                                    
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x7bbc6f]                                                                                                                                                                     
                                                                                                                                                                                                                                           
goroutine 94701 [running]:                                                                                                                                                                                                                 
main.(*SessionStatsCollector).Collect(0xc0000fe1e0, 0xc00b222000)                                                                                                                                                                          
        /go/src/github.com/metalmatze/transmission-exporter/cmd/transmission-exporter/session_stats_collector.go:118 +0xcf                                                                                                                 
github.com/prometheus/client_golang/prometheus.(*Registry).Gather.func1()                                                                                                                                                                  
        /go/pkg/mod/github.com/prometheus/client_golang@v0.9.0-pre1.0.20181010161331-7866eead363e/prometheus/registry.go:430 +0x19d                                                                                                        
created by github.com/prometheus/client_golang/prometheus.(*Registry).Gather                                                                                                                                                               
        /go/pkg/mod/github.com/prometheus/client_golang@v0.9.0-pre1.0.20181010161331-7866eead363e/prometheus/registry.go:522 +0xe12                                                                                                        
panic: runtime error: invalid memory address or nil pointer dereference                                                                                                                                                                    
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x7ba909]                                                                                                                                                                     
                                                                                                                                                                                                                                           
goroutine 94699 [running]:                                                                                                                                                                                                                 
main.(*SessionCollector).Collect(0xc0000ef880, 0xc00b222000)                                                                                                                                                                               
        /go/src/github.com/metalmatze/transmission-exporter/cmd/transmission-exporter/session_collector.go:134 +0xc9                                                                                                                       
github.com/prometheus/client_golang/prometheus.(*Registry).Gather.func1()                                                                                                                                                                  
        /go/pkg/mod/github.com/prometheus/client_golang@v0.9.0-pre1.0.20181010161331-7866eead363e/prometheus/registry.go:430 +0x19d                                                                                                        
created by github.com/prometheus/client_golang/prometheus.(*Registry).Gather                                                                                                                                                               
        /go/pkg/mod/github.com/prometheus/client_golang@v0.9.0-pre1.0.20181010161331-7866eead363e/prometheus/registry.go:441 +0x57d                                                                                                        
```

please make it so that if the exporter cant connect to transmission it returns 0 metrics, but dont crash.

--- a/cmd/transmission-exporter/session_collector.go
+++ b/cmd/transmission-exporter/session_collector.go
@@ -126,6 +126,7 @@ func (sc *SessionCollector) Collect(ch chan<- prometheus.Metric) {
 	session, err := sc.client.GetSession()
 	if err != nil {
 		log.Printf("failed to get session: %v", err)
+		return
 	}
 
 	ch <- prometheus.MustNewConstMetric(

--- a/cmd/transmission-exporter/session_stats_collector.go
+++ b/cmd/transmission-exporter/session_stats_collector.go
@@ -110,6 +110,7 @@ func (sc *SessionStatsCollector) Collect(ch chan<- prometheus.Metric) {
 	stats, err := sc.client.GetSessionStats()
 	if err != nil {
 		log.Printf("failed to get session stats: %v", err)
+		return
 	}
 
 	ch <- prometheus.MustNewConstMetric(

--- a/transmission.go
+++ b/transmission.go
@@ -3,6 +3,7 @@ package transmission
 import (
 	"bytes"
 	"encoding/json"
+	"errors"
 	"io/ioutil"
 	"net/http"
 	"strings"
@@ -46,7 +47,11 @@ func (c *Client) post(body []byte) ([]byte, error) {
 	}
 	defer res.Body.Close()
 
-	if res.StatusCode == 409 {
+	if res.StatusCode == http.StatusUnauthorized {
+		return make([]byte, 0), errors.New("authorization failed, check your username and password and make sure the ip is whitelisted")
+	}
+
+	if res.StatusCode == http.StatusConflict {
 		c.getToken()
 		authRequest, err := c.authRequest("POST", body)
 		if err != nil {

